lib/rules/no-constant-condition.js

```diff
@@ -144,10 +206,18 @@ module.exports = {
 
         return {
             ConditionalExpression: checkConstantCondition,
+            "ConditionalExpression:exit": checkConstantConditionLoopInSet,
             IfStatement: checkConstantCondition,
+            "IfStatement:exit": checkIfConstant,
```

Comment:
Should the `IfStatement` and `ConditionalExpression` listeners use `reportIfConstant` rather than `checkConstantCondition`? It seems like the `checkConstantCondition` function is only intended for loops.

Also, if I'm understanding correctly I think the `ConditionalExpression:exit` and `IfStatement:exit` handlers can be removed, since they don't need to be added to the set of loops.